pkg/privilege/privileges/cache.go

```diff
@@ -400,6 +411,121 @@ func (p *MySQLPrivilege) LoadAll(ctx sessionctx.Context) error {
 	return nil
 }
 
+func (p *immutable) loadSomeUsers(ctx sessionctx.Context, userList ...string) error {
+	err := p.loadTable(ctx, sqlLoadUserTable, p.decodeUserTableRow, userList...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = p.loadTable(ctx, sqlLoadGlobalPrivTable, p.decodeGlobalPrivTableRow, userList...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = p.loadTable(ctx, sqlLoadGlobalGrantsTable, p.decodeGlobalGrantsTableRow, userList...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = p.loadTable(ctx, sqlLoadDBTable, p.decodeDBTableRow, userList...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = p.loadTable(ctx, sqlLoadTablePrivTable, p.decodeTablesPrivTableRow, userList...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = p.loadTable(ctx, sqlLoadDefaultRoles, p.decodeDefaultRoleTableRow, userList...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	err = p.loadTable(ctx, sqlLoadColumnsPrivTable, p.decodeColumnsPrivTableRow, userList...)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	p.roleGraph = make(map[string]roleGraphEdgesTable)
+	err = p.loadTable(ctx, sqlLoadRoleGraph, p.decodeRoleEdgesTable)
+	if err != nil {
+		return errors.Trace(err)
+	}
+
+	return nil
+}
+
+func dedupSorted[S ~[]E, E any](s S, eq func(a, b E) bool) S {
```

Comment:
maybe rename to dedupKeepLast

Reply:
only works on sorted, so it's `dedupSortedKeepLast`

Reply:
done